File: alerts-services/mqtt/MqttPublisher.py
import json
from paho.mqtt.client import Client

class MqttPublisher:

    def __init__(self, client: Client, output_topic):
        self.client = client
        self.output_topic = output_topic

    def publish(self, payload):
        if isinstance(payload, dict):
            payload = json.dumps(payload)

        payload_bytes = payload.encode("utf-8")
        self.client.publish(self.output_topic, payload_bytes, qos=1)

File: sensors-mqtt/src/main/java/com/sistemasdistribuidos/mqtt/MqttPublisher.java
package com.sistemasdistribuidos.mqtt;

import org.eclipse.paho.client.mqttv3.MqttClient;
import org.eclipse.paho.client.mqttv3.MqttException;
import org.eclipse.paho.client.mqttv3.MqttMessage;

public class MqttPublisher {
    
    private final MqttClient client;
    
    public MqttPublisher(MqttClient client){
        this.client = client;
    }

    public void publish(String topic, String payload) throws MqttException{
        MqttMessage msg = new MqttMessage(payload.getBytes());
        msg.setQos(1);
        client.publish(topic, msg);
    }
}
